pkgs/os-specific/darwin/cctools/llvm.nix

```diff
@@ -0,0 +1,101 @@
+  # not appear to have issues, but the source is not available yet (as of June 2023).
+  useLLVMStrip = lib.versionAtLeast llvmVersion "15" || lib.versionAtLeast cctoolsVersion "1005.2";
+
+  llvm_bins = [
+    "dwarfdump"
+    "nm"
+    "objdump"
+    "size"
+    "strings"
+  ]
+  ++ lib.optional useLLVMBitcodeStrip "bitcode-strip"
+  ++ lib.optional useLLVMOtool "otool"
+  ++ lib.optional useLLVMStrip "strip";
+
+  # Only include the tools that LLVM doesn’t provide and that are present normally on Darwin.
+  # The only exceptions are the following tools, which should be reevaluated when LLVM is bumped.
+  # - install_name_tool (llvm-objcopy): unrecognized linker commands when building open source CF;
+  # - libtool (llvm-libtool-darwin): not fully compatible when used with xcbuild; and
+  # - lipo (llvm-lipo): crashes when running the LLVM test suite.
+  cctools_bins = [
+    "cmpdylib"
+    "codesign_allocate"
+    "ctf_insert"
+    "install_name_tool"
+    "ld"
+    "libtool"
+    "lipo"
+    "nmedit"
+    "pagestuff"
+    "ranlib"
+    "segedit"
+    "vtool"
+  ]
+  ++ lib.optional (!useLLVMBitcodeStrip) "bitcode_strip"
+  ++ lib.optional (!useLLVMOtool) "otool"
+  ++ lib.optional (!useLLVMStrip) "strip";
+
+  inherit (stdenv.cc) targetPrefix;
+in
+stdenv.mkDerivation {
+  pname = "cctools-llvm";
+  version = "${llvmVersion}-${cctoolsVersion}";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  # The `man` output has to be included unconditionally because dariwn.binutils expects it.
+  outputs = [ "out" "dev" "man" ];
+
+  buildCommand = ''
+    mkdir -p "$out/bin" "$man"
+    ln -s ${lib.getDev cctools-port} "$dev"
+
+    ${lib.optionalString enableManpages ''
+      man1="$man/share/man/man1"
+      mkdir -p "$man1"
+
+      function linkManPage() {
+        if [ -f "$1/share/man/man1/$2.1.gz" ]; then
+          ln -s "$1/share/man/man1/$2.1.gz" "$man1/''${3:-$2}.1.gz"
+        fi
+      }
+    ''}
+
+    # Use the clang-integrated assembler instead of using `as` from cctools.
+    makeWrapper "${lib.getBin llvmPackages.clang-unwrapped}/bin/clang" "$out/bin/${targetPrefix}as" \
+      --add-flags "-x assembler -integrated-as -c"
+
+    ln -s "${lib.getBin llvmPackages.bintools-unwrapped}/bin/llvm-ar" "$out/bin/${targetPrefix}ar"
+    ${lib.optionalString enableManpages ''linkManPage "${llvmPackages.llvm-manpages}" "llvm-ar" "ar"''}
+
+    for tool in ${toString llvm_bins}; do
+      cctoolsTool=''${tool/-/_}
+      ln -s "${lib.getBin llvmPackages.llvm}/bin/llvm-$tool" "$out/bin/${targetPrefix}$cctoolsTool"
+      ${lib.optionalString enableManpages ''linkManPage "${llvmPackages.llvm-manpages}" "llvm-$tool" "$cctoolsTool"''}
+    done
+
+    for tool in ${toString cctools_bins}; do
+      ln -s "${lib.getBin cctools-port}/bin/${targetPrefix}$tool" "$out/bin/${targetPrefix}$tool"
+      ${lib.optionalString enableManpages ''linkManPage "${lib.getMan cctools-port}" "$tool"''}
```

Comment:
I understand making the LLVM man pages optional but maybe the cctools man pages don't need to be. Those don't need to be built, right?

Or do you consider them not all that useful?

Reply:
There’s no harm in including them unconditionally. I was just trying to keep the semantics of `enableManpages` consistent by having it include either all or no man pages.

Let me know if I should make that change, though it wouldn’t affect users since building the man pages is the default. It’s only the stdenv that won’t build them (because it doesn’t need them).